Compile with explicitly specified path first

yousong opened the pull request:
```
Problem:    Wrong header and library files from system directory may be
            used instead of the ones specified on configure command line
Solution:   Compile with XXX_FLAGS appearing before ALL_CFLAGS
```

I came across the issue firstly with lua support when I try to have my own build of vim and its dependencies (https://github.com/yousong/build-scripts/blob/master/build-vim.sh).

nuko8 commented:
Just to make sure.  Does `$(call variable, param, param,…)` work with `make(1)` other than GNU make?

yousong committed:
Compile with explicitly specified path first

Problem:    Wrong header and library files from system directory may be
            used instead of the ones specified on configure command line
Solution:   Compile with XXX_FLAGS appearing before ALL_CFLAGS

yousong commented:
Other choices would be
1. Prepending those `LUA_CFLAGS`, `PERL_CFLAGS`, etc. to `ALL_CFLAGS`
2. Expand manually the `$(call CCC,xxx)` to eliminate possible compatibility issues altoghter

Personally I prefer the first choice, but it may also happen that different `XXX_CFLAGS` interfere with each other...

yousong commented:
Changes v1 -> v2
- Avoid using `$(call CCC,xxx)`
- Expand the change to include targets missed out in previous version

yousong commented:
Hi, @brammool , 3 monthes have passed, any thoughts on this one?
